add new module: moc

lasers opened the pull request:
**`moc`** is a console audio player designed to be powerful and easy to use.

This is similar to `cmus`. Two things I'm pondering about...

~~**No. 1**~~
~~I have `cache_timeout` and `cache_timeout_off`. The second one is for sleeping so it
does not run every 2s when the software is not running.~~

~~If we do this style, I wonder if it's a good idea to merge this instead of two separate
options.~~
~~* `cache_timeout = 5, 20` ---> (refresh `5`, sleep `20`)~~
~~* `cache_timeout = 5` ---> (refresh `5`, sleep `5`)  <--- How most modules usually
operate.~~

**No. 2**
Same thing for buttons. The number of buttons available for `cmus` and `moc` easily outnumber the number of buttons on most comfort mouses. We stick with this or same scenario with `No .1` above using a dictionary? One button config to rule them all.

We input what we want.
`button = {'pause': 1, 'stop': 3, 'volume_up': 4, 'volume_down': 5}`

Thanks for looking. This module was fun to put together.

`py3status` can always use more song players. 🎶  EDIT: I did ~~`No. 1` and~~ `No. 2`.

ultrabug commented:
Thanks for this @lasers 
I guess moc is not compatible with MPRIS...

Button 1, by pause you mean `play` or `toggle` ?

I think n°2 is okay with volume up/down unless it means that you actually want to handle amixer calls (then I think the volume module is enough)

lasers commented:
@ultrabug Yes. Button 1 = `mocp --toggle-pause` // `# Toggle between playing and paused`
The volume buttons does not touch alsa or pulse. It changes the volume in the client/server itself.

> I guess moc is not compatible with MPRIS...

Last think I checked weeks ago, it is not. There might be some moc/mocp plugins/scripts that could work with mpris. I could check them out later. 

Some new changes.

* I removed `button_toggle` + `button_toggle_cmd` here. We lose the ability to toggle `autonext`,
`repeat`, and/or `shuffle` together at once... in favor of KISS (Keep It Simple, Stupid)
options.

* Fixed `TypeError` when scrolling too fast with `seek_forward` / `seek_backward` by going with `data.get('state')` instead.

* Went with `cache_timeout = [5, 60]` here. I feel it's ready to be merged. Let me know if you want me to squash the commits. Thanks (unless you don't like the new cache_timeout or the new button).

ultrabug commented:
Yeah no I'm sorry but I'm not keen on those complex types on config options

lasers commented:
The `button = {dict}` is broken into an array of `button_*` and `cache_timeout = []` into two config options (`cache_timeout` and `sleep_timeout`).

This should be good to go. Speaking of removing config options to keep things simple. All buttons in `moc/cmus` can be easily removed by suggesting users to use `on_click` instead.

Please let me know if you need anything else. Many thanks! 👍

ultrabug commented:
> All buttons in moc/cmus/others can be easily removed by suggesting users to use on_click instead.

That's what I had in mind as well, the functionality is there already. Can you drop'em all extra buttons and also on cmus module ?

Then same comment as cmus, plz dont use sleep_timeout?

lasers commented:
@ultrabug This too.

ultrabug commented:
Thanks
